File: src/common/Smi.Common/Messages/Extraction/ExtractedFileStatusMessage.cs
namespace Smi.Common.Messages.Extraction
{
    /// <summary>
    /// Status message sent by services which extract files (CTP, FileCopier)
    /// </summary>
    public class ExtractedFileStatusMessage : ExtractMessage, IFileReferenceMessage
    {
        /// <summary>
        /// Original file path
        /// </summary>
        public string DicomFilePath { get; set; }

        /// <summary>
        /// The <see cref="ExtractedFileStatus"/> for this file
        /// </summary>
        public ExtractedFileStatus Status { get; set; }

        /// <summary>
        /// Output file path, relative to the extraction directory. Only required if an output file has been produced
        /// </summary>
        public string OutputFilePath { get; set; }

        /// <summary>
        /// Message required if Status is not 0
        /// </summary>
        public string StatusMessage { get; set; }


        public ExtractedFileStatusMessage() { }

        public ExtractedFileStatusMessage(ExtractFileMessage request)
            : base(request)
        {
            DicomFilePath = request.DicomFilePath;
            OutputFilePath = request.OutputPath;
        }

        public override string ToString() =>
            $"{base.ToString()}," +
            $"DicomFilePath={DicomFilePath}," +
            $"ExtractedFileStatus={Status}," +
            $"OutputFilePath={OutputFilePath}," +
            $"StatusMessage={StatusMessage}," +
            "";
    }
}

File: src/microservices/com.smi.microservices.ctpanonymiser/src/main/java/org/smi/ctpanonymiser/messages/ExtractedFileStatusMessage.java
package org.smi.ctpanonymiser.messages;

import org.smi.common.messageSerialization.JsonDeserializerWithOptions.FieldRequired;
import org.smi.common.messages.ExtractMessage;
import org.smi.common.messages.IMessage;
import org.smi.ctpanonymiser.util.ExtractedFileStatus;

/**
 * Message indicating the path to an anonymised file
 */
public class ExtractedFileStatusMessage extends ExtractMessage implements IMessage {

	@FieldRequired
	public String DicomFilePath;

	@FieldRequired
	public String OutputFilePath;

    @FieldRequired
	public ExtractedFileStatus Status;

	@FieldRequired
	public String StatusMessage;

	/** @return the filePath */
	public String getFilePath() {
		return DicomFilePath;
	}

	/**
	 * @param filePath
	 *            the filePath to set
	 */
	public void setFilePath(String dicomFilePath) {
		DicomFilePath = dicomFilePath;
	}

	/** @return the Project Number */
	public String getProjectNumber() {
		return ProjectNumber;
	}

	/**
	 * @param projectNumber
	 *            the projectNumber to set
	 */
	public void setProjectNumber(String projectNumber) {
		ProjectNumber = projectNumber;
	}

	public ExtractedFileStatusMessage(ExtractFileMessage request) {

		ExtractionJobIdentifier = request.ExtractionJobIdentifier;
		ExtractionDirectory = request.ExtractionDirectory;
        Modality = request.Modality;
		DicomFilePath = request.DicomFilePath;
		ProjectNumber = request.ProjectNumber;
		JobSubmittedAt = request.JobSubmittedAt;
	}

	@Override
	public String toString() {

		StringBuilder sb = new StringBuilder();

		sb.append("ExtractionJobIdentifier: " + ExtractionJobIdentifier + "\n");
		sb.append("ExtractionDirectory: " + ExtractionDirectory + "\n");
		sb.append("Modality: " + Modality + "\n");
		sb.append("DicomFilePath: " + DicomFilePath + "\n");
		sb.append("ProjectNumber: " + ProjectNumber + "\n");
		sb.append("JobSubmittedAt: " + JobSubmittedAt + "\n");
		sb.append("OutputFilePath: " + OutputFilePath + "\n");
		sb.append("Status: " + Status + "\n");
		sb.append("StatusMessage: " + StatusMessage + "\n");

		return sb.toString();
	}

}
